### ISSUE TITLE
DOM elements move around during SSR

### ISSUE THREAD

**Issue by adam-kov** [sveltejs/kit#7711]
### Describe the bug

I have a list of items to display like so:
```svelte
<ul>
  {#each items as item}
    <li>
      <a href={getPath(item)} class="block">
        <div class="content">...</div>
      </a>
    </li>
  {/each}
</ul>
```
However during SSR the `div` that has the content pops outside of the `a` tag and the HTML sent to the browser looks like this:
```html
<ul>
  <li>
    <a href="/something" class="block"></a>
    <div class="content">...</div>
  </li>
</ul>
```
After hydration, the content placed back inside the `a` tag correctly, but this causes a lot of shifting and misplacement during loading.
The problem is with the `a` tag, because removing it solves the shifting, but that's not a feasable option for us.

### Reproduction

~https://stackblitz.com/edit/sveltejs-kit-template-default-w7ajcm?file=src/lib/list.svelte~

### System Info
WIth WSL
```Shell
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 AMD Ryzen 7 5800X 8-Core Processor
    Memory: 17.23 GB / 31.89 GB
  Binaries:
    Node: 16.14.2 - C:\Program Files\nodejs\node.EXE
    npm: 8.12.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (107.0.1418.42)    
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @sveltejs/adapter-node: next => 1.0.0-next.85
    @sveltejs/adapter-static: ^1.0.0-next.32 => 1.0.0-next.38
    @sveltejs/kit: next => 1.0.0-next.403
    svelte: ^3.44.0 => 3.49.0
```

### Severity

annoyance

**Comment by dominikg** [comment]
could you clarify what you mean with "during ssr" and where exactly you see the invalid structure ? the ssr result sent to the browser looks like this on your stackblitz (obtained using the network debugger, document response to `https://sveltejs-kit-template-default-w7ajcm--5173.local-credentialless.webcontainer.io/`)

```html
<a href="/bar" class="block s-vgILjrO1zWzW"><div class="content s-vgILjrO1zWzW">bar</div></a>
```
div inside a, not below a.


```html
<!DOCTYPE html>
<html lang="en">
	<head>
		<script src="/.localservice@runtime.2ef3dd6d2178aae5fbf423082bd20f2467c7ac49.js"></script>
		<meta charset="utf-8" />
		<link rel="icon" href="./favicon.png" />
		<meta name="viewport" content="width=device-width" />
		
	<style data-sveltekit>ul.s-vgILjrO1zWzW{list-style:none;margin:0;padding:0}.block.s-vgILjrO1zWzW{display:block;background-color:#dedeee}.content.s-vgILjrO1zWzW{padding:20px 40px;margin-bottom:10px;color:#000}.s-vgILjrO1zWzW{}</style>
	</head>
	<body data-sveltekit-prefetch>
		


<ul class="s-vgILjrO1zWzW"><li class="s-vgILjrO1zWzW"><a href="/foo" class="block s-vgILjrO1zWzW"><div class="content s-vgILjrO1zWzW">foo</div></a>
    </li><li class="s-vgILjrO1zWzW"><a href="/bar" class="block s-vgILjrO1zWzW"><div class="content s-vgILjrO1zWzW">bar</div></a>
    </li><li class="s-vgILjrO1zWzW"><a href="/baz" class="block s-vgILjrO1zWzW"><div class="content s-vgILjrO1zWzW">baz</div></a>
    </li>
</ul>


		<script type="module" data-sveltekit-hydrate="16ti0mc">
                 // snipped for brevity
		</script>
	
	</body>
</html>

```

**Comment by adam-kov** [comment]
Yea, sorry, the reproduction doesn't really show the problem.

I took screenshots of the app during loading (document received but not hydrated) and after everything is settled:

**Loading**
![image](https://user-images.githubusercontent.com/43071496/202683385-8ee03f47-ca5b-45c6-90b0-9062b7503374.png)

**Settled**
![image](https://user-images.githubusercontent.com/43071496/202683500-f80996a3-30a3-4082-9bed-c76ac3f13914.png)

**Comment by dominikg** [comment]
It's great that you provided a stackblitz, however it needs to reproduce the issue for us to be able to analyze it. If it is only reproducible outside of stackblitz, please share a link to a minimal github repo.

**Comment by Rich-Harris** [comment]
I downloaded the StackBlitz repro to run locally and I can't reproduce the issue — the `<div>` is appearing inside the `<a>` even when I disable JavaScript. 

That said, there are some constraints in HTML that SvelteKit can't enforce at SSR time — in theory, flow content can't contain block content, and the HTML parser will move things around. In practice, block content inside flow content is perfectly fine as long as you construct it programmatically, which is why hydration reverts it to its expected state.

I'd be curious to see a reproduction that shows the issue, but I'm not sure it's something that it's in our power to fix.

**Comment by adam-kov** [comment]
Dear all, thanks for the quick responses. As I was trying to create a reproduction by stripping down the production app I found what caused my issue. It has **nothing** to do with Svelte and I'll close the issue, mea culpa.

*For those who might need this info later:*
As it turns out, nesting `a` tags inside eachother is not valid HTML, so the parser will try to unnest them and that caused the elements to pop outside their parent tags. Today I learned.
